pkg/auth/authproviders/backend_factory.go

```diff
@@ -8,7 +8,7 @@ import (
 // BackendFactory is responsible for creating Backends.
 type BackendFactory interface {
 	// CreateBackend creates a new backend instance for the given auth provider, using the specified configuration.
-	CreateBackend(ctx context.Context, id string, uiEndpoints []string, config map[string]string) (Backend, error)
+	CreateBackend(ctx context.Context, id string, uiEndpoints []string, config map[string]string, mappings map[string]string) (Backend, error)
```

Comment:
How about creating a struct for this params? Maybe a custom type instead of string?